plugins/modules/user_role_workflow_manager.py

```diff
@@ -1599,53 +1598,59 @@ def create_user(self, user_params):
-        for key in required_keys:
-            if key not in user_params:
-                missing_keys.append(key)
+            required_keys = ['username', 'password']
+            missing_keys = []
 
-        if missing_keys:
-            error_message = "Mandatory parameter(s) '{0}' not present in the user details.".format(", ".join(missing_keys))
-            return {"error_message": error_message}
+            self.log("Check if each required key is present in the user_params dictionary...", "DEBUG")
+            for key in required_keys:
+                if key not in user_params:
+                    missing_keys.append(key)
 
-        try:
-            response = self.dnac._exec(
-                family="user_and_roles",
-                function="add_user_api",
-                op_modifies=True,
-                params=user_params,
-            )
-            self.log("Received API response from create_user: {0}".format(str(response)), "DEBUG")
-            self.created_user.append(user_params.get("username"))
-            return response
+            if missing_keys:
+                error_message = "Mandatory parameter(s) '{0}' not present in the user details.".format(", ".join(missing_keys))
+                return {"error_message": error_message}
 
-        except Exception as e:
-            self.log("Unexpected error occurred: {0}".format(str(e)), "ERROR")
-            if "[403]" in str(e):
-                error_message = (
-                    "The Catalyst Center user '{0}' does not have the necessary permissions to 'create or update' a user through the API.".format(
-                        self.payload.get("dnac_username"))
+            try:
+                response = self.dnac._exec(
+                    family="user_and_roles",
+                    function="add_user_api",
+                    op_modifies=True,
+                    params=user_params,
                 )
-            else:
-                error_message = "Invalid email format for email '{0}' under username '{1}'".format(user_params.get("email"), user_params.get("username"))
+                self.log("Received API response from create_user: {0}".format(str(response)), "DEBUG")
+                self.created_user.append(user_params.get("username"))
+                return response
 
-            return {"error_message": error_message}
+            except Exception as e:
+                self.log("Unexpected error occurred: {0}".format(str(e)), "ERROR")
+                if "[403]" in str(e):
+                    error_message = (
+                        "The Catalyst Center user '{0}' does not have the necessary permissions to 'create or update' a user through the API.".format(
+                            self.payload.get("dnac_username"))
+                    )
+                else:
+                    error_message = "Invalid email format for email '{0}' under username '{1}'".format(user_params.get("email"), user_params.get("username"))
```

Comment:
error_message = "Invalid email format for '{0}' associated with username '{1}'.".format(
    user_params.get("email"), user_params.get("username")
)